// @sijanbhattarai/mdx-editor-maps/vite.config.ts
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import { resolve } from 'path';

export default defineConfig({
  plugins: [react()],
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.ts'),
      name: 'MDXEditorMaps',
      formats: ['es'],
      fileName: (format) => `index.js`
    },
    rollupOptions: {
      external: [
        'react',
        'react-dom',
        '@mdxeditor/editor',
        '@teamimpact/veda-ui',
        '@tanstack/react-query',
        'react-router-dom',
        '@sijanbhattarai/mdx-editor-core',
        '@sijanbhattarai/mdx-editor-utils'
      ],
      output: {
        globals: {
          react: 'React',
          'react-dom': 'ReactDOM'
        }
      }
    },
    target: 'esnext',
    sourcemap: true
  },
  resolve: {
    alias: {
      '@sijanbhattarai/mdx-editor-core': resolve(__dirname, '../mdx-editor-core/src'),
      '@sijanbhattarai/mdx-editor-utils': resolve(__dirname, '../mdx-editor-utils/src')
    }
  }
});
