// @nasa-impact/mdx-editor-utils/vite.config.ts
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import { resolve } from 'path';

export default defineConfig({
  plugins: [react()],
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.ts'),
      name: 'MDXEditorUtils',
      formats: ['es'],
      fileName: (format) => `index.js`
    },
    rollupOptions: {
      external: [
        'react',
        'react-dom',
        '@mdxeditor/editor',
        'unified',
        'remark-parse',
        'remark-stringify',
        'mdast-util-from-markdown',
        'mdast-util-mdx-jsx',
        'micromark-extension-mdx-jsx',
        'unist-util-visit'
      ],
      output: {
        // Preserve the directory structure in dist
        preserveModules: true,
        preserveModulesRoot: 'src',
        entryFileNames: '[name].js',
        // Provide globals for external dependencies
        globals: {
          react: 'React',
          'react-dom': 'ReactDOM'
        }
      }
    },
    target: 'esnext',
    sourcemap: true
  }
});
